Cython/Compiler/ExprNodes.py

```diff
@@ -14038,7 +14248,7 @@ def calculate_result_code(self):
         return self.arg.result_as(self.type)
 
     def generate_result_code(self, code):
-        self.arg.generate_result_code(code)
+        pass
```

Comment:
Why was this wrong for a cast? Are you sure there is never anything to do for `self.arg` here?

Reply:
`arg` is a subexpr. So therefore `arg.generate_result_code(code)` gets called in `ExprNode.generate_evaluation_code` -> `ExprNode.generate_subexpr_evaluation_code`. That means that `arg.generate_result_code` ends up being called twice.

(Possibly makes sense to cherry-pick this fix out and apply it separately rather than waiting for this larger PR though)